### PR TITLE
Fix common package installation errors

### PR DESCRIPTION
**Pull request by amcasey**
- If there's a yarn.lock in the root, prefer yarn, unless there's a package-lock.json
- Pass --no-immutable to yarn2 to override CI=true
- Pass --no-frozen-lockfile to pnpm to override CI=true
- Skip a bunch of unmanageable repos

Bonus: improve error output

- yarn and pnpm tend to mix stderr and stdout, so print both
- Drop all `npm WARN` lines, rather than trying to identify special `ENOSPC` lines (whackamole style)

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
How do you currently know if we should be using yarn 1, 2, or 3? Do the newer ones subsume the older ones?

**Comment by DanielRosenwasser** [comment]
> isRepoYarn2 

ah

File: src/main.ts [hunk 1]
```diff
@@ -51,6 +53,13 @@ const skipRepos = [
     "https://github.com/BabylonJS/Babylon.js", // Runs out of space during compile
     "https://github.com/eclipse-theia/theia", // Probably same
     "https://github.com/wbkd/react-flow", // Probably same
+    "https://github.com/remix-run/remix", // Too big to fit on VM
+    "https://github.com/NervJS/taro", // Too big to fit on VM
+    "https://github.com/TanStack/table", // Too big to fit on VM
+    "https://github.com/doczjs/docz", // Too big to fit on VM
+    "https://github.com/NativeScript/NativeScript", // Uses NX package manager
+    "https://github.com/wulkano/Kap", // Incompatible with Linux
+    "https://github.com/lit/lit", // Depends on non-public package
```
Comment (DanielRosenwasser) [inline src/main.ts:62]: Huh, which one?

File: src/main.ts [hunk 2]
```diff
@@ -437,30 +447,9 @@ function sanitizeErrorText(text: string): string {
...
-        }
-
-        result += message.substring(index, newIndex);
-        result += newStr;
-
-        index = newIndex + oldStr.length;
-    }
+    // These are uninteresting in general and actually problematic when there are
+    // thousands of instances of ENOSPC (which also appears as an error anyway)
+    return message.replace(/npm WARN.*\n/g, "");
```
Comment (DanielRosenwasser) [inline src/main.ts:452]: Is this per line? Should it be anchored to the beginning of the line (and is there some extra prefix you need if so)? Do you want `.replaceAll`?

**Comment by DanielRosenwasser** [comment]
> - Pass --no-immutable to yarn2 to override CI=true
> - Pass --no-frozen-lockfile to pnpm to override CI=true

So we intentionally want to run with newer dependencies, right?

**Review by DanielRosenwasser — APPROVED** [review]

File: src/main.ts (hunk 1 above)
Comment (amcasey) [reply src/main.ts:62]: @lit-internal/something

**Comment by amcasey** [comment]
> How do you currently know if we should be using yarn 1, 2, or 3? Do the newer ones work on the older ones?

Is there a yarn 3?  There's one repo that rejects `--mode:skip-build` in favor of `--skip-build` and I couldn't figure out why.  Didn't seem important, so I didn't investigate.  Yarn 2 we detect based on additional config files.

AFAIK, yarn 1 and yarn 2 are mutually incompatible, chiefly because they take different arguments.

**Comment by amcasey** [comment]
> > * Pass --no-immutable to yarn2 to override CI=true
> > * Pass --no-frozen-lockfile to pnpm to override CI=true
> 
> So we intentionally want to run with newer dependencies, right?

I didn't see a `ci` equivalent for yarn or pnpm, so I assumed this was just a sign that some people had failed to keep their lockfiles up-to-date.  If you know of one, I'm happy to switch.

**Pull request merged**